src/WorkspaceDependencyViewExtension/WorkspaceDependencyViewExtension.cs

```diff
@@ -89,7 +91,13 @@ public void Loaded(ViewLoadedParams viewLoadedParams)
 
             pmExtension.PackageLoader.PackgeLoaded += (package) =>
             {
-                DependencyView.DependencyRegen(viewLoadedParams.CurrentWorkspaceModel as WorkspaceModel);
+                DynamoModel model = viewLoadedParams.dynamoViewModel.Model;
+                WorkspaceModel workspace = model.CurrentWorkspace;
+                String filePath = Path.GetFullPath(workspace.FileName);
+                model.ClearCurrentWorkspace();
+                model.OpenFileFromPath(filePath, false);
+
+              //  DependencyView.DependencyRegen(viewLoadedParams.CurrentWorkspaceModel as WorkspaceModel);
```

Comment:
So I checked; if this is triggered by the `PackageLoaded` event this code will indeed execute before all the libraries are reloaded and the VM is reset, which is not what we want.

Reply:
I imagine a step being added after this line here: https://github.com/DynamoDS/Dynamo/blob/499bd9b359fdd52bd977b2e7b5ca1113ca63e80a/src/DynamoCore/Engine/EngineController.cs#L505 )when all libraries have been reloaded after importing a new one) to deserialize/resolve the dummy nodes again. It could be a new event or something else.